doc/developer/design/20230814_unbilled_replicas.md

```diff
@@ -0,0 +1,166 @@
+# Unbilled replicas
+
+- [#20317)](https://github.com/MaterializeInc/materialize/issues/20317)
+
+## Context
+
+Support requires the ability to dynamically add resources to a user's environment
+without incurring substantial costs for the user. For example, when support
+notices that after a version upgrade, Materialize needs additional compute
+resources, support should be able to add such resources for a limited amount of
+time without requiring explicit approval and complicated reimbursement for the
+customer.
+
+We extend Materialize with a concept of unbilled replicas, which server compute
+traffic but are not charged to a customer's account.
+
+## Goals
+
+We introduce unbilled clusters by the following means:
+* Introduce cluster profiles.
+* Add a billing property to cluster profiles.
+* Record cluster profile properties in audit events.
+
+## Non-Goals
+
+We do not plan to do the following:
+* Allow customers general access to cluster profiles.
+
+## Overview
+
+We introduce the notion of cluster profiles, which capture groups
+of replica sharing configuration aspects. Profiles can co-exist, and a default
+profile captures the current configuration.
```

Comment:
> current configuration

This is misleading because it can't describe the current configuration of the cluster, but only of replicas not in any profile, right?